kubeadm/scripts/Install-Containerd.ps1

```diff
@@ -0,0 +1,158 @@
+        exit 1
+    }
+}
+
+function CalculateSubNet {
+    param (
+        [string]$gateway,
+        [int]$prefixLength
+    )
+    $len = $prefixLength
+    $parts = $gateway.Split('.')
+    $result = @()
+    for ($i = 0; $i -le 3; $i++) {
+        if ($len -ge 8) {
+            $mask = 255
+
+        }
+        elseif ($len -gt 0) {
+            $mask = ((256 - 2 * (8 - $len)))
+        }
+        else {
+            $mask = 0
+        }
+        $len -= 8
+        $result += ([int]$parts[$i] -band $mask)
+    }
+
+    $subnetIp = [string]::Join('.', $result)
+    $cidr = 32 - $prefixLength
+    return "${subnetIp}/$cidr"
+}
+
+$requiredWindowsFeatures = @(
+    "Containers",
+    "Hyper-V",
+    "Hyper-V-PowerShell")
+
+function ValidateWindowsFeatures {
+    $allFeaturesInstalled = $true
+    foreach ($feature in $requiredWindowsFeatures) {
+        $f = Get-WindowsFeature -Name $feature
+        if (-not $f.Installed) {
+            Write-Warning "Windows feature: '$feature' is not instealled."
+            $allFeaturesInstalled = $false
+        }
+    }
+    return $allFeaturesInstalled
+}
+
+if (-not (ValidateWindowsFeatures)) {
+    Write-Output "Installing required windows features..."
+
+    foreach ($feature in $requiredWindowsFeatures) {
+        Install-WindowsFeature -Name $feature
+    }
+
+    Write-Output "Please reboot and re-run this script."
+    exit 0
+}
+
+Write-Output "Getting ContainerD binaries"
+$global:ConainterDPath = "$env:ProgramFiles\containerd"
+mkdir -Force $global:ConainterDPath | Out-Null
+DownloadFile "$global:ConainterDPath\containerd.tar.gz" https://github.com/containerd/containerd/releases/download/v${ContainerDVersion}/containerd-${ContainerDVersion}-windows-amd64.tar.gz
+tar.exe -xvf "$global:ConainterDPath\containerd.tar.gz" --strip=1 -C $global:ConainterDPath
+$env:Path += ";$global:ConainterDPath"
+[Environment]::SetEnvironmentVariable("Path", $env:Path, [System.EnvironmentVariableTarget]::Machine)
+containerd.exe config default | Out-File "$global:ConainterDPath\config.toml" -Encoding ascii
+#config file fixups
+$config = Get-Content "$global:ConainterDPath\config.toml"
+$config = $config -replace "bin_dir = (.)*$", "bin_dir = `"c:/opt/cni/bin`""
+$config = $config -replace "conf_dir = (.)*$", "conf_dir = `"c:/etc/cni/net.d`""
+$config | Set-Content "$global:ConainterDPath\config.toml" -Force 
+
+mkdir -Force c:\opt\cni\bin | Out-Null
+mkdir -Force c:\etc\cni\net.d | Out-Null
+
+Write-Output "Getting SDN CNI binaries"
+DownloadFile "c:\opt\cni\cni-plugins.zip" https://github.com/microsoft/windows-container-networking/releases/download/v0.2.0/windows-container-networking-cni-amd64-v0.2.0.zip
+Expand-Archive -Path "c:\opt\cni\cni-plugins.zip" -DestinationPath "c:\opt\cni\bin" -Force
+
+mkdir -Force c:\k | Out-Null
+DownloadFile "c:\k\hns.psm1" https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/hns.psm1
+Import-Module "c:\k\hns.psm1"
+New-HnsNetwork -Type NAT -Name nat
```

Comment:
Is `New-HnsNetwork -Type NAT -Name nat` the same as `docker network create -d nat host` in prepare-node? It looks like the nat network is configured in both scripts in different ways.  

Is there a reason the network config for the nat needs to be done here and not in prepare node?

Reply:
They should be effectively the same.
Let me move this to PrepareNode.ps1